src/geom/stroke.rs

```diff
@@ -59,15 +146,124 @@ impl<T: Debug> Debug for PartialStroke<T> {
+    }
+}
+
+cast_from_value! {
+    DashPattern: "dash pattern",
+    // Use same names as tikz:
+    // https://tex.stackexchange.com/questions/45275/tikz-get-values-for-predefined-dash-patterns
+    "solid" => Vec::new().into(),
+    "dotted" => vec![DashLength::LineWidth, Abs::pt(2.0).into()].into(),
+    "densely dotted" => vec![DashLength::LineWidth, Abs::pt(1.0).into()].into(),
+    "loosely dotted" => vec![DashLength::LineWidth, Abs::pt(4.0).into()].into(),
+    "dashed" => vec![Abs::pt(3.0).into(), Abs::pt(3.0).into()].into(),
+    "densely dashed" => vec![Abs::pt(3.0).into(), Abs::pt(2.0).into()].into(),
+    "loosely dashed" => vec![Abs::pt(3.0).into(), Abs::pt(6.0).into()].into(),
+    "dashdotted" => vec![Abs::pt(3.0).into(), Abs::pt(2.0).into(), DashLength::LineWidth, Abs::pt(2.0).into()].into(),
+    "densely dashdotted" => vec![Abs::pt(3.0).into(), Abs::pt(1.0).into(), DashLength::LineWidth, Abs::pt(1.0).into()].into(),
+    "loosely dashdotted" => vec![Abs::pt(3.0).into(), Abs::pt(4.0).into(), DashLength::LineWidth, Abs::pt(4.0).into()].into(),
+    array: Vec<DashLength> => {
+        Self {
+            dash_array: array.into_iter().collect(),
+            dash_phase: Length::zero(),
+        }
+    },
+    mut dict: Dict => {
+        let dash_array: Vec<DashLength> = dict.take("array")?.cast()?;
+        let dash_phase = dict.take("phase").ok().map(Length::cast)
+            .transpose()?.unwrap_or(Length::zero());
+
+        dict.finish(&["array", "phase"])?;
+
+        Self {
+            dash_array,
+            dash_phase,
+        }
+    },
+}
+
+impl Resolve for DashPattern {
+    type Output = DashPattern<Abs>;
+
+    fn resolve(self, styles: StyleChain) -> Self::Output {
+        DashPattern {
+            dash_array: self.dash_array.into_iter()
+                .map(|l| l.resolve(styles)).collect(),
+            dash_phase: self.dash_phase.resolve(styles),
+        }
+    }
+}
+
 cast_from_value! {
     PartialStroke: "stroke",
     thickness: Length => Self {
         paint: Smart::Auto,
         thickness: Smart::Custom(thickness),
+        line_cap: Smart::Auto,
+        line_join: Smart::Auto,
+        dash_pattern: Smart::Auto,
+        miter_limit: Smart::Auto,
     },
     color: Color => Self {
         paint: Smart::Custom(color.into()),
         thickness: Smart::Auto,
+        line_cap: Smart::Auto,
+        line_join: Smart::Auto,
+        dash_pattern: Smart::Auto,
+        miter_limit: Smart::Auto,
+    },
+    mut dict: Dict => {
+        fn take<T: Cast<Value>>(dict: &mut Dict, key: &str) -> StrResult<Smart<T>> {
+            Ok(dict.take(key).ok().map(T::cast)
+                .transpose()?.map(Smart::Custom).unwrap_or(Smart::Auto))
+        }
+
+        let paint = take::<Paint>(&mut dict, "color")?;
+        let thickness = take::<Length>(&mut dict, "thickness")?;
+        let line_cap = take::<LineCap>(&mut dict, "line_cap")?;
+        let line_join = take::<LineJoin>(&mut dict, "line_join")?;
```

Comment:
```suggestion
        let cap = take::<LineCap>(&mut dict, "cap")?;
        let join = take::<LineJoin>(&mut dict, "join")?;
```
maybe we can drop the `line_` prefix everywhere?